src/types.rs

```diff
@@ -226,22 +229,38 @@ impl PublicId {
         public_key : id.get_public_key(),
         public_sign_key : id.get_public_sign_key(),
         validation_token : id.get_validation_token(),
-        name : id.get_name()
+        name : id.get_name(),
+        original_name : id.get_original_name(),
       }
     }
 
+    pub fn name(&self) -> NameType {
+      self.name.clone()
+    }
```

Comment:
Can we do it without the implicit cloning? In many places we just need the reference.